v2/client/client.go

```diff
@@ -241,13 +240,9 @@ func (c *ceClient) StartReceiver(ctx context.Context, fn interface{}) error {
 					respFn = noRespFn
 				}
 
-				if err == io.EOF { // Normal close
-					return
-				}
-
 				if err != nil {
-					cecontext.LoggerFrom(ctx).Warn("Error while receiving a message: ", err)
-					continue
+					errChan <- err
```

Comment:
not all errors cause all protocols to not respond, some would like the message to be returned via Invoke, this change does not seem like it will work for all protocols.

Is there a change to the amqp impl that would allow it to handle the error? Like, perhaps it needs to respond with a `io.EOF` when the the server is disconnected.

Reply:
Thanks for the review.

I think it does make send to send `io.EOF` in case connection is down, I'll look in that direction.